mcs/class/Mono.Profiler.Log/Mono.Profiler.Log/LogProcessor.cs

```diff
@@ -109,6 +109,8 @@ LogEvent ReadEvent ()
 				case LogEventType.AllocationBacktrace:
 				case LogEventType.AllocationNoBacktrace:
 					return new AllocationEvent {
+						Buffer = _bufferHeader,
```

Comment:
Instead of all this repetition, just add a `LogEvent ev` local variable, assign to it in all cases instead of immediately returning here, and set the properties before returning `ev`.

Reply:
yes, or maybe as parameters to the base LogEvent constructor? Thus we "force" every Event class instance to provide those

Reply:
That ends up with the same amount of repetition, just elsewhere. Since `LogEvent` instances can only be constructed by the library (by virtue of the `LogEvent` constructor being `internal`), I think it's fine to just use the `internal` setters on `Buffer` and `Timestamp`.